src/charm.py

```diff
@@ -74,12 +75,28 @@ def __init__(self, *args):
         self.cassandra = Cassandra(charm=self)
         logging.getLogger("cassandra").setLevel(logging.CRITICAL)
 
+    def config_okay(self, event: EventBase) -> bool:
+        """Ensure required config values are provided and valid."""
+        if "heap_size" not in self.config:
+            self.unit.status = BlockedStatus('Config value "heap_size" required')
+            event.defer()
+            return False
+        heap_size = self.config["heap_size"]
+        if match(r"^\d+[KMG]$", heap_size, IGNORECASE) is None:
```

Comment:
According to config.yaml, lowercase multipliers are accepted too.

```suggestion
        if match(r"^\d+[KkMmGg]$", heap_size, IGNORECASE) is None:
```

Reply:
Done.